pkg/controller/instanceset/in_place_update_util.go

```diff
@@ -259,7 +259,10 @@ func equalBasicInPlaceFields(old, new *corev1.Pod) bool {
 
 func equalResourcesInPlaceFields(old, new *corev1.Pod) bool {
 	if len(old.Spec.Containers) != len(new.Spec.Containers) {
```

Comment:
I think we can simply remove this if-statement to support the case.

The following for-loop will make sure that all containers in the `new` pod are present in the `old` pod and have the same resources, so if a new container is added to the new pod, the function returns false, which causes a pod recreation. But if the old pod contains unknown containers (injected by external webhooks), the function will still return true, which causes an in-place update.

Reply:
great suggestion!